core/src/main/java/org/apache/iceberg/actions/BinPackStrategy.java

```diff
@@ -155,9 +155,12 @@ public RewriteStrategy options(Map<String, String> options) {
   public Iterable<List<FileScanTask>> planFileGroups(Iterable<FileScanTask> dataFiles) {
     ListPacker<FileScanTask> packer = new BinPacking.ListPacker<>(maxGroupSize, 1, false);
     List<List<FileScanTask>> potentialGroups = packer.pack(dataFiles, FileScanTask::length);
+
     return potentialGroups.stream().filter(group ->
-      group.size() >= minInputFiles || sizeOfInputFiles(group) > targetFileSize ||
-              group.stream().anyMatch(this::taskHasTooManyDeletes)
+            (group.size() >= minInputFiles && group.size() > 1) ||
+                sizeOfInputFiles(group) > targetFileSize ||
```

Comment:
I think this would be easier to understand if was > maxFileSize although realistically at this point in the code it could only get to this point if the size was larger than maxFileSize.